code/game/objects/effects/decals/decal.dm

```diff
@@ -4,12 +4,10 @@
 	layer = TURF_DECAL_LAYER
 
 /obj/effect/turf_decal/Initialize(mapload)
-	. = ..()
-	qdel(src) // return INITIALIZE_HINT_QDEL <-- Doesn't work
-
-/obj/effect/turf_decal/ComponentInitialize()
-	. = ..()
+	..()
+	. = INITIALIZE_HINT_QDEL
 	var/turf/T = loc
 	if(!istype(T)) //you know this will happen somehow
-		CRASH("Turf decal initialized in an object/nullspace")
+		stack_trace("Turf decal initialized in an object/nullspace")
+		return
```

Comment:
Why convert CRASH() to stack_trace() return?

Reply:
So it can still return `INITIALIZE_HINT_QDEL`.

Reply:
`CRASH()` still returns the `.` value
![image](https://user-images.githubusercontent.com/25063394/185071514-905257b5-d019-4ed3-b482-1ad37463fcec.png)

Reply:
Neat, didn't know that. Changing it back.